<p>This rule detects computing the scalar product of a vector with itself</p>
<h2>Noncompliant Code Example</h2>
<pre>
VectorMultiplier multiplier;
multiplier.computeScalarProduct(v, v);
</pre>
<h2>Compliant Code Example</h2>
<pre>
VectorMultiplier multiplier;
multiplier.computeSquaredNorm(v);
</pre>
